pkg/reconciler/testing/v1/roles.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2024 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	v1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// createRole creates a Role in the Kubernetes cluster.
+func CreateRole(namespace, roleName, verbs, resources, apiGroup string) *v1.Role {
```

Comment:
I think that instead of passing all these strings in as parameters to this function, we should follow the pattern of passing in functions that modify the role before returning the final role.